extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java

```diff
@@ -331,19 +331,45 @@ ServiceStartBuildItem startLiquibase(LiquibaseRecorder recorder,
         return Collections.emptySet();
     }
 
-    private Optional<String> extractChangeFile(Change change) {
+    private Optional<String> extractChangeFile(Change change, String changeSetFilePath) {
+        String path = null;
+        Boolean relative = null;
         if (change instanceof LoadDataChange) {
-            return Optional.ofNullable(((LoadDataChange) change).getFile());
+            LoadDataChange loadDataChange = (LoadDataChange) change;
+            path = loadDataChange.getFile();
+            relative = loadDataChange.isRelativeToChangelogFile();
+        } else if (change instanceof SQLFileChange) {
+            SQLFileChange sqlFileChange = (SQLFileChange) change;
+            path = sqlFileChange.getPath();
+            relative = sqlFileChange.isRelativeToChangelogFile();
+        } else if (change instanceof CreateProcedureChange) {
+            CreateProcedureChange createProcedureChange = (CreateProcedureChange) change;
+            path = createProcedureChange.getPath();
+            relative = createProcedureChange.isRelativeToChangelogFile();
+        } else if (change instanceof CreateViewChange) {
+            CreateViewChange createViewChange = (CreateViewChange) change;
+            path = createViewChange.getPath();
+            relative = createViewChange.getRelativeToChangelogFile();
         }
-        if (change instanceof SQLFileChange) {
-            return Optional.ofNullable(((SQLFileChange) change).getPath());
+
+        // unrelated change or change does not reference a file (e.g. inline view)
+        if (path == null) {
+            return Optional.empty();
         }
-        if (change instanceof CreateProcedureChange) {
-            return Optional.ofNullable(((CreateProcedureChange) change).getPath());
+        // absolute file path or changeSet has no file path
+        if (relative == null || !relative || changeSetFilePath == null) {
+            return Optional.of(path);
         }
-        if (change instanceof CreateViewChange) {
-            return Optional.ofNullable(((CreateViewChange) change).getPath());
+
+        // relative file path needs to be resolved against changeSetFilePath
+        // notes: ClassLoaderResourceAccessor does not provide a suitable method and CLRA.getFinalPath() is not visible
+        int idxOfLastSlash = changeSetFilePath.lastIndexOf('/');
```

Comment:
Shouldn't we use `Path.resolve()` instead of doing some dark magic?

Reply:
In the beginning, I actually thought about that as well, but then got distracted by the `ClassLoaderResourceAccessor` approach (which didn't work) and forgot about it.
I'll push an update.

Reply:
Ah, now I remember: On Windows I would then need to replace `\` with `/`.

Reply:
Still better than doing dark magic :).

Reply:
I agree that it's a mess anyway given this: https://github.com/liquibase/liquibase/blob/v4.2.2/liquibase-core/src/main/java/liquibase/resource/ClassLoaderResourceAccessor.java#L121 (which is also not accessible easily)

Problem resolved.